testutil/network/basenet/network.go

```diff
@@ -0,0 +1,106 @@
+package basenet
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdknetwork "github.com/cosmos/cosmos-sdk/testutil/network"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/testkeyring"
+)
+
+var _ network.InMemoryCosmosNetwork = (*BaseInMemoryCosmosNetwork)(nil)
+
+// BaseInMemoryCosmosNetwork is an "abstract" (i.e. partial) implementation, intended
+// to be embedded by other ("concrete") InMemoryCosmosNetwork implementations.
+type BaseInMemoryCosmosNetwork struct {
+	Config               network.InMemoryNetworkConfig
+	PreGeneratedAccounts *testkeyring.PreGeneratedAccountIterator
+	Network              *sdknetwork.Network
+
+	lastAccountSeqNumber int32
```

Comment:
```suggestion
	lastAccountSeqNumber atomic.int32
```
If we later use it atomically to get the next why not have it as an atomic the whole time

Reply:
Nice! I keep forgetting that's a thing. :sweat_smile:

Reply:
I think, in this case, it actually makes things harder.